action.yml

```diff
@@ -0,0 +1,154 @@
+name: 'Build, test, package'
+description: 'Build, test, and package distributions for an ITK external module.'
+inputs:
+  cmake-build-type:
+    required: false
+    default: 'MinSizeRel'
+  cmake-options:
+    description: 'Additional CMake arguments to use in building and testing the current remote module.'
+    required: false
+    default: ''
+  itk-cmake-options:
+    description: 'Additional CMake arguments to use in building ITK, such as enabling certain remote modules.'
+    required: false
+    default: ''
+  itk-git-tag:
+    required: false
+    default: '835dc01388d22c4b4c9a46b01dbdfe394ec23511' # v5.3rc04.post2
+
+runs:
+  using: "composite"
+  steps:
+    - name: Define OS-dependent environment variables
+      if: runner.os == 'Windows'
+      shell: bash
+      run: |
+        echo "C_COMPILER=cl.exe" >> $GITHUB_ENV       # defines env.C_COMPILER
+        echo "CXX_COMPILER=cl.exe" >> $GITHUB_ENV     # defines env.CXX_COMPILER
```

Comment:
I was thinking, e.g.

```
if test -z "${{ inputs.c-compliler }}"; then
       echo "C_COMPILER=cl.exe" >> $GITHUB_ENV       # defines env.C_COMPILER
else
       echo "C_COMPILER={{ inputs.c-compiler }}" >> $GITHUB_ENV       # defines env.C_COMPILER
fi
```

etc.

Reply:
Thanks for the example, I hadn't thought about doing the OS string checks as well in a single step as opposed to one step per OS. I'll move forward with the current state for now but I'm open to revisiting in the future if demand for compiler input specification arises.